src/commands/moderator/LoggingSync.ts

```diff
@@ -0,0 +1,104 @@
+import {BaseCommand, ICommandContext, ICommandInfo} from "../BaseCommand";
+import {MongoManager} from "../../managers/MongoManager";
+import {MessageUtilities} from "../../utilities/MessageUtilities";
+import {Logger} from "../../utilities/Logger";
+
+const LOGGER: Logger = new Logger(__filename, false);
+export class LoggingSync extends BaseCommand {
+    private _isRunning: boolean = false;
+
+    public constructor() {
+        const cmi: ICommandInfo = {
+            cmdCode: "LOGGING_SYNC_COMMAND",
+            formalCommandName: "Logging Sync",
+            botCommandName: "loggingsync",
+            description: "Updates list of users who have logged items for this guild.",
+            rolePermissions: [
+                "Moderator"
+            ],
+            generalPermissions: [],
+            botPermissions: [],
+            commandCooldown: 15 * 60 * 1000,
+            argumentInfo: [],
+            guildOnly: true,
+            botOwnerOnly: false
+        };
+
+        super(cmi);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public async run(ctx: ICommandContext): Promise<number> {
+        if (this._isRunning) {
+            await ctx.interaction.reply({
+                content: "This command is currently in use. Please wait a bit.",
+                ephemeral: true
+            });
+
+            return -1;
+        }
+
+        const guild = ctx.guild!;
+        const guildDoc = ctx.guildDoc!;
+        const usersWithLogs = guildDoc.properties.usersWithLogs ?? [];
+        const startingCount = usersWithLogs.length;
+
+        this._isRunning = true;
+
+        let embed = MessageUtilities.generateBlankEmbed(ctx.guild!, "RANDOM")
+            .setTitle("Preparing to Sync.")
+            .setDescription(`Started with ${startingCount} members with logs.\nFetching all members in this server...`)
+            .setFooter({text: "This might take a while, please wait."})
+            .setTimestamp();
+
+        await ctx.interaction.reply({
+            embeds: [embed]
+        });
+
+        const allMembers = await guild.members.fetch()
```

Comment:
Are you sure we want to fetch every member like this?

Reply:
Yes, this is only run a single time per guild.